app/commands/notification/create.rb

```diff
@@ -6,10 +6,10 @@ class Create
 
     def call
       klass = "notifications/#{type}_notification".camelize.constantize
-      klass.create!(
-        user: user,
-        params: params
-      )
+
+      klass.create!(user: user, params: params).tap do
```

Comment:
For another discussion, but should we enforce whether commands have a return value or not?

Reply:
Yeah - I was thinking about this. 

Two options:
1. All `create` commands return something
2. We rely on the `on_success`/`on_failure` states.

If we do (1) then errors should raise Exceptions. If we do (2) then errors can be passed bac.

The main downside to (2) is that it makes both the internal and consuming code a bit more verbose:

```
  res = Notification::Create.(...) 
  handle_error unless res.success?
  notification = res.result
```

But that verbosity is also maybe slightly clearer. Thoughts?

Reply:
I personally like (2) as it's the most unsurprising option. Running the `call` method for me implies doing an action and not returning anything.